src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -107,10 +118,27 @@ private BindingDataResolver GetResolver(PropertyInfo propInfo, INameResolver nam
                 return GetAppSettingResolver((string)originalValue, appSettingAttr, nameResolver, propInfo);
             }
             // try to resolve with auto resolve ({...}, %...%)
-            if (autoResolveAttr != null && originalValue != null)
+            if (autoResolveAttr != null)
             {
-                _autoResolves[propInfo] = autoResolveAttr;
-                return GetTemplateResolver((string)originalValue, autoResolveAttr, nameResolver, propInfo, contract);
+                if (propInfo.PropertyType != typeof(string))
```

Comment:
can you move this logic into its own GetAutoResolveResolver function? I think it improves readability & is better with single responsibility